layer_analyzer.go

```diff
@@ -0,0 +1,124 @@
+package lifecycle
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/lifecycle/buildpack"
+	"github.com/buildpacks/lifecycle/cmd"
+	"github.com/buildpacks/lifecycle/launch"
+	"github.com/buildpacks/lifecycle/platform"
+)
+
+type LayerAnalyzer interface {
+	Analyze(buildpacks []buildpack.GroupBuildpack, skipLayers bool, appMeta platform.LayersMetadata, cache Cache) (platform.CacheMetadata, error)
+}
+
+type DefaultLayerAnalyzer struct {
+	Logger            Logger
+	LayersDir         string
+	MetadataRetriever CacheMetadataRetriever
+	Platform          cmd.Platform
+}
+
+func NewLayerAnalyzer(logger Logger, metadataRetriever CacheMetadataRetriever, layersDir string, platform cmd.Platform) LayerAnalyzer {
+	return &DefaultLayerAnalyzer{
+		LayersDir:         layersDir,
+		Logger:            logger,
+		MetadataRetriever: metadataRetriever,
+		Platform:          platform,
+	}
+}
+
+func (la *DefaultLayerAnalyzer) Analyze(buildpacks []buildpack.GroupBuildpack, skipLayers bool, appMeta platform.LayersMetadata, cache Cache) (platform.CacheMetadata, error) {
```

Comment:
I wonder if we could come up with a better name for this interface. "Layer analyzer" makes me think it's only reading from the app image, but it's doing so much more!

Reply:
Actually if we do https://github.com/buildpacks/lifecycle/pull/582#discussion_r615109465 (or something like it) this concern goes away...

Reply:
I am +1 on finding a new name for this. I'll start with some suggestions

- `LayerMetadataRestorer` (My favorite, re-reading the code)
- `LayerMetadataProcessor`

Reply:
I like `LayerMetadataRestorer`!